### PR TITLE
Added random param gen tests. Added device awareness for parameter generators.

### PR DESCRIPTION
**Pull request by shijianjian**
### Description
Will be continuously working on this PR.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

**Review by edgarriba — APPROVED** [review]

File: test/augmentation/test_random_generator_3d.py [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+    def test_valid_param_combinations(
+        self, depth, height, width, distortion_scale, batch_size, same_on_batch, device, dtype
+    ):
+        random_perspective_generator3d(
+            batch_size=batch_size, depth=depth, height=height, width=width,
+            distortion_scale=distortion_scale.to(device=device, dtype=dtype), same_on_batch=same_on_batch)
+
+    @pytest.mark.parametrize('depth,height,width,distortion_scale', [
+        # Should be failed if distortion_scale > 1. or distortion_scale < 0.
+        pytest.param(-100, 100, 100, torch.tensor(0.5), marks=pytest.mark.xfail),
```
Comment (edgarriba) [inline test/augmentation/test_random_generator_3d.py:45]: why `pytest.mark.fail` instead of `pytest.raises` ?
as in of https://github.com/kornia/kornia/blob/master/test/color/test_gray.py#L24

Comment (shijianjian) [reply test/augmentation/test_random_generator_3d.py:45]: ```pytest.mark.xfail``` is recommended by the doc when working with parametrize: https://docs.pytest.org/en/stable/parametrize.html

Comment (edgarriba) [reply test/augmentation/test_random_generator_3d.py:45]: is this a potential fail or always fail ? I see you can pass to `pytest.raises` too.

Comment (shijianjian) [reply test/augmentation/test_random_generator_3d.py:45]: Those functions are always expected failing. We cannot pass ```pytest.raises``` to marks.
